callouts/python/extproc/service/callout_tools.py

```diff
@@ -140,23 +141,25 @@ def add_body_mutation(
     clear_body: bool = False,
     clear_route_cache: bool = False,
 ) -> BodyResponse:
-  """Generate a body response for incoming requests.
+  """Generate a body mutation response for incoming requests.
+  
+    If both body and clear_body are left as default, the incoming request's body will not be modified.
 
-Args:
-  body: Text of the body.
-  clear_body: If set to true, the modification will clear the previous body,
-    if left false, the text will be appended to the end of the previous
-    body.
-  clear_route_cache: If true, will enable clear_route_cache on the response.
+  Args:
+    body: Body text to replace the current body of the incomming request.
+    clear_body: If true, will clear the body of the incomming request. 
+    clear_route_cache: If true, will enable clear_route_cache on the response.
```

Comment:
why do we have clear_route_cache for responses? thats useless isnt it?

Reply:
If we do not clear the route cache on a request body no-op response unless the envoy is configured to clear the route cache every time, A no-op request callout will cause the response callout to not be processed in every case. This is mainly for the request body as that is the one that is being processed before the response. It would be impractical to make one method for request bodies and one for response bodies.

Reply:
@tyxia is this a bug?

Reply:
This is not the expected behavior.
First, clear_route_cache is only applicable to mainstream **request** path. Second, I don't follow the logic why clear_route_cache on request path could affect response path behavior

@jstraceski I think this should require more debugging

Reply:
So, the reason why the function says `clear_route_cache on the response` is because the returning datatype of this function is a `BodyResponse`, I have made a quick patch to make this more evident. This is the first pass I have done just on the callout_tools.py file, and some associated changes.